modules/avd/avd.bicep

```diff
@@ -119,3 +134,192 @@ resource workspace 'Microsoft.DesktopVirtualization/workspaces@2019-12-10-previe
 // }
 
 output appGroupName string = appGroupName
+
+
+
+// // Azure Virtual Desktop and Session Hosts region
+
+resource nic 'Microsoft.Network/networkInterfaces@2020-06-01' = [for i in range(0, countAVDInstances): {
+  name: 'nic-redcap-${i}'
+  location: location
+  properties: {
+    ipConfigurations: [
+      {
+        name: 'ipconfig'
+        properties: {
+          privateIPAllocationMethod: 'Dynamic'
+          subnet: {
+            id: ComputeSubnetId
+          }
+        }
+      }
+    ]
+  }
+}]
+
+resource vm 'Microsoft.Compute/virtualMachines@2023-03-01' = [for i in range(0, countAVDInstances): {
+  name: 'vm-redcap-${i}'
+  location: location
+  properties: {
+    licenseType: 'Windows_Client'
+    hardwareProfile: {
+      vmSize: vmSku
+    }
+    osProfile: {
+      computerName: 'vm-redcap-${i}'
+      adminUsername: avdVMAdmin
+      adminPassword: avdVMPassword
+      windowsConfiguration: {
+        enableAutomaticUpdates: false
+        patchSettings: {
+          patchMode: 'Manual'
+        }
+      }
+    }
+    storageProfile: {
+      osDisk: {
+        name: 'vm-OS-${i}'
+        caching: vmDiskCachingType
+        managedDisk: {
+          storageAccountType: vmDiskType
+        }
+        osType: 'Windows'
+        createOption: 'FromImage'
+      }
+      // TODO Turn into params
+      imageReference: {
+        publisher: 'microsoftwindowsdesktop'
+        offer: 'office-365'
+        sku: '20h2-evd-o365pp'
+        version: 'latest'
+      }
+      dataDisks: []
+    }
+    networkProfile: {
+      networkInterfaces: [
+        {
+          id: nic[i].id
+        }
+      ]
+    }
+  }
+  dependsOn: [
+    nic[i]
+  ]
+}]
+
+// Reference https://github.com/Azure/avdaccelerator/blob/e247ec5d1ba5fac0c6e9f822c4198c6b41cb77b4/workload/bicep/modules/avdSessionHosts/deploy.bicep#L162
```

Comment:
This is where the trouble begins. I have a lot of commented out info from all that I tried so far

Reply:
@matheuscarboni Revert to using API version `2019-12-10-preview`.

Reply:
I've tried setting the host pool, the `getHostPool` and the `reference` to 2019-12-10-preview, and 2021-01-14-preview to no avail. Going to keep digging to see what else can be done
![image](https://github.com/kalalvishal/azure-redcap-paas/assets/20691499/b86596b9-b3bb-4452-be55-f80fb53a3e29)